### ISSUE TITLE
How add protocol in options?

### ISSUE THREAD

**Issue by fghpdf** [cyrus-and/chrome-remote-interface#99]
The source code shows the protocol urls.
![image](https://cloud.githubusercontent.com/assets/7081582/25070279/f5691d42-22cc-11e7-9e2d-ffc6bb2d4e6a.png)

I downloaded them and want to offer them by options, but I don`t know how to do it.

My version is V8, Is that mean I should offer two protocol ?
![image](https://cloud.githubusercontent.com/assets/7081582/25070306/7898bbdc-22cd-11e7-810f-bbdf342c6887.png)

By the way, I used the `remote: true`

Thanks!

**Comment by cyrus-and** [comment]
> The source code shows the protocol urls.

You should probably ignore that URLs. Chromium team has decided to split the protocol in two parts to separate the *V8* part from the *browser* part; you really need both.

---

If you want to fetch the protocol for a given Chrome version you can just run:

    chrome-remote-interface protocol -r > protocol.json

Then if you want to use that protocol version:

```js
const options = {
    protocol: JSON.parse(fs.readFileSync('protocol.json'));
};
```

Hope this helps.

**Comment by fghpdf** [comment]
Thanks to your answer.

But I got this protocol that seems not json, it`s js file
![image](https://cloud.githubusercontent.com/assets/7081582/25071113/f59e68a2-22e1-11e7-85ce-fe4a512e7e73.png)
My os is debian and use dev chrome headless mode

**Comment by cyrus-and** [comment]
Oh yes, my bad, the bundled utility should output proper JSON. I'm gonna fix it right away...

**Comment by cyrus-and** [comment]
Here we go, now `chrome-remote-interface protocol -r > protocol.json` should give you a proper JSON file to use as I previously said.

Please let me know if this solves your issue.

**Comment by fghpdf** [comment]
I`m sorry about that this doesn`t slove my issue.

This is my commond and some info
![image](https://cloud.githubusercontent.com/assets/7081582/25077556/ca897274-22f0-11e7-944a-cce7fb77cfdf.png)

And I got this file

[download file](https://drive.google.com/open?id=0B8G0yE0uaAf9RlZmYUE3MWMyanc)

the file md5 is `9c5193f2763781ac3a4fd1307d1a7e46`

I can convert it to json by tools, do you know whether result is right or not.

My Chrome version is
59.0.3067.0 dev version (64 bit)
and javascript is
V8 5.9.195

May this info can help you.

May be I am wrong.

First, I use
`google-chrome-unstable --headless --remote-debugging-port=9222 --disable-gpu
`
to open headless chrome

ps: last time I add `--remote-debugging-address=192.168.1.138` ,this time I removed it because I found I can`t connect by localhost
ps 2: maybe I should `-h 192.168.1.138`,is it?

Secondly, I use `chrome-remote-interface protocol -r -> protocol.json` and got that result.

If you want more info, I`m glad to show you.

My english is very pool,sorry......:)

**Comment by cyrus-and** [comment]
> Im sorry about that this doesnt slove my issue.

Are you sure that you've updated to the latest version after myfix? You need to run:

    npm update -g chrome-remote-interface

You can check the current version with:

    npm show chrome-remote-interface version

> ps: last time I add --remote-debugging-address=192.168.1.138 ,this time I removed it because I found I cant connect by localhost ps 2: maybe I should-h 192.168.1.138`,is it?

Well if you just specify the port, Chrome will listen on `localhost`. I don't think that the `--remote-debugging-address` option actually works, despite the description:

> Use the given address instead of the default loopback for accepting remote debugging connections. Should be used together with --remote-debugging-port. Note that the remote debugging protocol does not perform any authentication, so exposing it too widely can be a security risk.

If I set that option Chrome still listens on `localhost`.

**Comment by fghpdf** [comment]
My bad, I forget update....

Thank you very much!

**Comment by mohyeid** [comment]
I have the other way where I want chrome to listen to localhost not 127.0.0.1. Any help?

**Comment by cyrus-and** [comment]
@mohyeid localhost usually resolves to 127.0.0.1, what are you really trying to do?
